src/Chapter3.hs

```diff
@@ -1006,6 +1248,143 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+newtype HealthPoints = HealthPoints
+    { healthPoints    :: Int
+    } deriving (Show)
+
+newtype DefencePoints = DefencePoints
+    { defencePoints   :: Int
+    } deriving (Show)    
+
+data KnightAttack
+    = KnightAttacking 
+    | Heal 
+    | IncreaseDefence 
+    deriving(Show, Enum, Bounded, Eq, Ord)
+
+data MonsterAttack
+    = MonsterAttacking
+    | RunAway 
+    deriving(Show, Enum, Bounded, Eq, Ord)
+
+data Knight = Knight 
+    { knightHealth           :: HealthPoints,
+      knightDefence          :: DefencePoints,
+      knightsAttackPower     :: Int,
+      knightsLastAttack      :: Maybe KnightAttack
+    } deriving(Show)
+
+testKnight :: Knight
+testKnight = Knight { knightHealth = HealthPoints { healthPoints = 9 }, knightDefence = DefencePoints { defencePoints = 3}, knightsAttackPower = 5, knightsLastAttack = Nothing}
+
+data Monster = Monster
+    { monsterHealth           :: HealthPoints,
+      monstersAttackPower     :: Int,
+      monstersLastAttack      :: Maybe MonsterAttack
+    } deriving(Show)
+
+testMonster :: Monster
+testMonster = Monster { monsterHealth = HealthPoints { healthPoints = 24 }, monstersAttackPower = 1, monstersLastAttack = Nothing}
+
+testMonster2 :: Monster
+testMonster2 = Monster { monsterHealth = HealthPoints { healthPoints = 15 }, monstersAttackPower = 13, monstersLastAttack = Nothing}
+
+data Warrior
+    = WarMonster { warMonster :: Monster }
```

Comment:
Could you please clarify, why this type is necessary? I see that it is a wrapper around the `Monster` 🤔